encoding.bs

```diff
@@ -94,37 +96,177 @@ bitwise AND by "&amp;", and bitwise OR by "|".
+
+<p>To read a number <var>number</var> of <a>items</a> from <var>ioQueue</var>,
+run these steps:
+
+<ol>
+ <li><p>Let <var>readItems</var> be an empty list.
+
+ <li>
+  <p>Perform the following step <var>number</var> times:
+
+  <ol>
+   <li><p><a for=list>Append</a> to <var>readItems</var> the result of
+   <a for="I/O queue">reading</a> an item from <var>ioQueue</var>.
+  </ol>
+ </li>
+
+ <li><p><a for=list>Remove</a> <a>end-of-file</a> from <var>readItems</var>.
+
+ <li><p>Return <var>readItems</var>.
+</ol>
+
+<p>To <dfn id=concept-stream-push for="I/O queue" export>push</dfn> one
+<a>item</a> <var>item</var> to an <a for=/>I/O queue</a> <var>ioQueue</var>, run
+these steps:
 
-<p>When one or more <a>tokens</a> are
-<dfn id=concept-stream-prepend for=stream>prepended</dfn> to a
-<a for=/>stream</a>, those tokens must be inserted, in given order,
-before the first token in the stream.
+<ol>
+ <li><p>If the last <a>item</a> in <var>ioQueue</var> is <a>end-of-file</a>,
+ then <a for=list>insert</a> <var>item</var> before the last <a>item</a> in the
+ queue.
+
+ <li><p>Otherwise, <a for=list>append</a> <var>item</var> to <var>ioQueue</var>.
+</ol>
 
-<p class=example id=example-tokens>Inserting the sequence of tokens <code>&amp;#128169;</code>
-in a stream "<code> hello world</code>", results in a stream
-"<code>&amp;#128169; hello world</code>". The next token to be read would be
-<code>&amp;</code>. <!-- 💩 -->
+<p>To push a sequence of items is to push each item in the sequence, in the
+given order.
 
-<p>When one or more <a>tokens</a> are
-<dfn id=concept-stream-push for=stream>pushed</dfn> to a <a for=/>stream</a>,
-those tokens must be inserted, in given order, after the last token in the stream.
+<p>To <dfn id=concept-stream-prepend for="I/O queue">prepend</dfn> one
+<a>item</a> to an <a for=/>I/O queue</a>, perform the normal <a>list</a>
+<a for=list>prepend</a> operation. To prepend a sequence of items, insert those
+items, in the given order, before the first item in the queue.
 
+<p class=example id=example-items>Inserting the sequence of scalar value items
+<code>&amp;#128169;</code> in an I/O queue of scalar values "<code> hello world</code>",
+results in an I/O queue "<code>&amp;#128169; hello world</code>". The next item
+to be read would be <code>&amp;</code>. <!-- 💩 -->
+
+<p class=note><a for=/>I/O queues</a> are defined as <a>lists</a>, not
+<a spec=infra>queues</a>, because they feature a <a for="I/O queue">prepend</a>
+operation. However, this prepend operation is an internal detail of the
+algorithms in this specification, and is not to be used by other standards.
+Implementations are free to find alternative ways to implement such algorithms,
+as detailed in [[#implementation-considerations]].
+
+<hr>
+
+<p><a for=/>I/O queues</a> can be implicitly converted into and from
+<a>strings</a> and <a>byte sequences</a> using the following operations.
+
+<p>To <dfn for="byte sequence to I/O queue" export>convert</dfn> a <a>byte
```

Comment:
If we're going to define this as part of this PR we should also reference it where it's needed below I think.